modal/network_file_system.py

```diff
@@ -343,12 +342,16 @@ def gen_transfers():
                 relpath_str = subpath.relative_to(_local_path).as_posix()
                 yield subpath, PurePosixPath(remote_path, relpath_str)
 
-        transfer_paths = aiostream.stream.iterate(gen_transfers())
-        await aiostream.stream.map(
-            transfer_paths,
-            aiostream.async_(lambda paths: self.add_local_file(paths[0], paths[1], progress_cb)),
-            task_limit=20,
-        )
+        async with aclosing(sync_or_async_iter(gen_transfers())) as transfer_paths:
+            async with aclosing(
+                async_map(
+                    transfer_paths,
+                    lambda paths: self.add_local_file(paths[0], paths[1], progress_cb),
+                    concurrency=20,
+                )
+            ) as stream:
+                async for item in stream:
+                    await item
```

Comment:
The returned item will be the return value of the awaited `self.add_local_file` which is just an int we don't really care about - you can just iterate the stream and `pass` instead (to make it execute the full mapping)